### ISSUE TITLE
About Data Discriminator Loss

### ISSUE THREAD

**Issue by heixiaoniu** [kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound#1]
Hello, first of all, thank you very much for providing the code.
I would like to ask, the coefficient of D_Loss in the paper is multiplied by a negative constant, but the actual setting in the code is positive 0.2, why is this? 
In addition, when I am training, if I use a negative constant, total_loss will be negative, and there is no way to optimize it in the end. Is there any solution?

**Comment by kaen2891** [comment]
Hey @heixiaoniu ,

The loss function for our Adversarial Fine-tuning is defined as follows:
```math
\mathcal{L}_{\text{Final}} = \mathcal{L}_{\text{CE}} + \lambda \, \!  \mathcal{L}_{\text{Dis}}
```

The $$\mathcal{L}_{Dis}$$ term gets multiplied by a negative constant, which you can see in [this line](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/models/ast.py#L256) of the code. For more details, I'd recommend checking out [this page](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/models/functions.py) or the DANN paper. It's a great resource for understanding this mechanism.

Second, about the 0.2 you mentioned, it seems like it's referring to the [alpha value](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/scripts/icbhi_aft_1msteps_500_mixed.sh#L30), right? This value is applied when calculating the [final loss is calculated](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/main.py#L384C41-L384C51). We did not explicitly explain this in the paper, though; we just replaced it with $$\lambda$$ in the above formula. To break it down, the detailed loss function actually looks like this:
```math
\mathcal{L}_{\text{Final}} = \mathcal{L}_{\text{CE}} + \lambda \, \! \alpha \, \! \mathcal{L}_{\text{Dis}}
```
Here, $$\alpha$$ is 0.2, and in [this line](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/main.py#L384), you'll see it as `args.alpha`. The $$\lambda$$ value gets calculated in [this line](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/main.py#L361) and then utilized in [this line](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/models/functions.py#L14C42-L14C47) as `cts.alpha`.

If you run our code using [this script](https://github.com/kaen2891/adversarial_fine-tuning_using_generated_respiratory_sound/blob/main/Classification/scripts/icbhi_aft_1msteps_500_mixed.sh) and set `--adversarial_ft`, everything should work smoothly. There's no need to manually set a negative constant for multiplying with the loss function. That's already taken care of in our implementations.

**Comment by heixiaoniu** [comment]
OK, Thank you very much!!!!
